plugins/flytekit-inference/flytekitplugins/inference/ollama/serve.py

```diff
@@ -0,0 +1,187 @@
+            base_code = """
+import base64
+import ollama
+"""
+            encoded_modelfile = base64.b64encode(self._model_modelfile.encode("utf-8")).decode("utf-8")
+
+            if "{inputs" in self._model_modelfile:
+                python_code = f"""
+{base_code}
+import os
+import json
+import sys
+
+from flyteidl.core import literals_pb2 as _literals_pb2
+from flytekit.core import utils
+from flytekit.core.context_manager import FlyteContextManager
+from flytekit.interaction.string_literals import literal_map_string_repr
+from flytekit.models import literals as _literal_models
+from flytekit.models.core.types import BlobType
+from flytekit.types.file import FlyteFile
+
+input_arg = sys.argv[-1]
+
+ctx = FlyteContextManager.current_context()
+local_inputs_file = os.path.join(ctx.execution_state.working_dir, 'inputs.pb')
+ctx.file_access.get_data(
+    input_arg,
+    local_inputs_file,
+)
+input_proto = utils.load_proto_from_file(_literals_pb2.LiteralMap, local_inputs_file)
+idl_input_literals = _literal_models.LiteralMap.from_flyte_idl(input_proto)
+
+inputs = literal_map_string_repr(idl_input_literals)
+
+for var_name, literal in idl_input_literals.literals.items():
+    if literal.scalar and literal.scalar.blob:
+        if (
+            literal.scalar.blob.metadata.type.dimensionality
+            == BlobType.BlobDimensionality.SINGLE
+        ):
+            downloaded_file = FlyteFile.from_source(literal.scalar.blob.uri).download()
+            inputs[var_name] = downloaded_file
+
+"""
+
+                python_code += """
+class AttrDict(dict):
+    'Convert a dictionary to an attribute style lookup. Do not use this in regular places, this is used for namespacing inputs and outputs'
+
+    def __init__(self, *args, **kwargs):
+        super(AttrDict, self).__init__(*args, **kwargs)
+        self.__dict__ = self
+
+
+inputs = {'inputs': AttrDict(inputs)}
+
+"""
+
+                python_code += f"""
+encoded_model_file = '{encoded_modelfile}'
+"""
+
+                python_code += """
+modelfile = base64.b64decode(encoded_model_file).decode('utf-8').format(**inputs)
+modelfile = modelfile.replace('{', '{{').replace('}', '}}')
+
+with open('Modelfile', 'w') as f:
+    f.write(modelfile)
+"""
+
+                python_code += f"""
+for chunk in ollama.create(model='{self._model_name}', path='Modelfile', stream=True):
+    print(chunk)
+"""
+            else:
+                python_code = f"""
+{base_code}
+
+encoded_model_file = '{encoded_modelfile}'
+
+modelfile = base64.b64decode(encoded_model_file).decode('utf-8')
+
+with open('Modelfile', 'w') as f:
+    f.write(modelfile)
+
+for chunk in ollama.create(model='{self._model_name}', path='Modelfile', stream=True):
+    print(chunk)
+"""
+        else:
+            python_code = f"""
+import ollama
+
+for chunk in ollama.pull('{self._model_name}', stream=True):
+    print(chunk)
+"""
+
+        command = (
+            f'sleep 15 && python3 -c "{python_code}" {{{{.input}}}}'
```

Comment:
is `{{{{.input}}}}` correct?

Reply:
yes. the parentheses shouldn't be escaped in the f-string. it is equivalent to `{{.input}}`.